plugins/modules/site_workflow_manager.py

```diff
@@ -787,81 +1356,229 @@ def get_diff_merged(self, config):
+                    )
+                    return {"error_message": error_message}
+
+            if not self.site_requires_update():
+                self.update_not_neeeded_sites.append(site_name)
+                self.log(
+                    "Site - {0} does not need any update".format(site_name), "INFO")
 
-        if site_created or site_updated:
-            if response and isinstance(response, dict):
+                return self
+
+            if site_updated and response and isinstance(response, dict):
                 executionid = response.get("executionId")
                 while True:
                     execution_details = self.get_execution_details(executionid)
                     if execution_details.get("status") == "SUCCESS":
                         self.result['changed'] = True
                         break
-
                     elif execution_details.get("bapiError"):
-                        self.module.fail_json(msg=execution_details.get("bapiError"),
-                                              response=execution_details)
+                        self.module.fail_json(msg=execution_details.get(
+                            "bapiError"), response=execution_details)
                         break
 
                 if site_updated:
                     self.updated_site_list.append(site_name)
-                    self.log("Site - {0} Updated Successfully".format(site_name), "INFO")
+                    self.log(
+                        "Site - {0} Updated Successfully".format(site_name), "INFO")
+
+                else:
+                    site_exists, current_site = self.site_exists()
+                    if site_exists:
+                        self.created_site_list.append(site_name)
+                        self.log(
+                            "Site '{0}' created successfully".format(site_name), "INFO")
+
+            elif self.dnac_version >= self.version_2_3_7_6 and self.site_requires_update():
+                site_params = self.want.get("site_params")
+                site_params["site_id"] = self.have.get("site_id")
+                site_type = site_params.get("type")
+
+                response = (self.update_floor(site_params) if site_type == "floor"
+                            else self.update_area(site_params) if site_type == "area"
+                            else self.update_building(site_params) if site_type == "building"
+                            else self.log("Unknown site type: {0}".format(site_type), "ERROR"))
+
+                if response:
+                    self.log("Received API response from 'update_site': {0}".format(
+                        str(response)), "DEBUG")
+                    site_updated = True
+
+                if site_updated and response and isinstance(response, dict):
+                    taskid = response["response"]["taskId"]
+                    task_details = self.get_task_details(taskid)
```

Comment:
You have to use the newer API 'Get Tasks by ID' , this get_task_by_id will going to deprecate soon.

Reply:
Changed to "get_tasks_by_id"